export/mlab_export_test.py

```diff
@@ -0,0 +1,492 @@
+    self.assertTrue(mock_rrd_export.called)
+
+  def testcover_init_global(self):
+    mlab_export.init_global()
+
+  @mock.patch('mlab_export.logging.error')
+  def testcover_parse_args(self, mock_error):
+    mlab_export.sys.argv = [
+        'mlab_export.py', '--bad-argument', '--causes-error']
+
+    self.assertRaises(SystemExit, mlab_export.parse_args, 0)
+
+    self.assertTrue(mock_error.called)
+
+  def testunit_align_timestamp(self):
+    fake_ts = 1410341339
+    expected_6 = fake_ts - 5
+    expected_10 = fake_ts - 9
+    expected_30 = fake_ts - 29
+
+    ts_6 = mlab_export.align_timestamp(fake_ts, 6)
+    ts_10 = mlab_export.align_timestamp(fake_ts, 10)
+    ts_30 = mlab_export.align_timestamp(fake_ts, 30)
+
+    self.assertEqual(ts_6, expected_6)
+    self.assertEqual(ts_10, expected_10)
+    self.assertEqual(ts_30, expected_30)
+
+  @mock.patch('mlab_export.time.time')
+  def testunit_default_end_time(self, mock_time):
+    mock_time.return_value = 1401401409
+    mock_options = mock.Mock()
+    mock_options.length = 3600
+    mock_options.step = 10
+    mock_options.update = False
+    expected_ts_end = 1401400800
+
+    returned_ts_end = mlab_export.default_end_time(mock_options)
+
+    self.assertEqual(returned_ts_end, expected_ts_end)
+
+  @mock.patch('mlab_export.time.time')
+  def testunit_default_end_time_WHEN_short_length_RAISES_TimeOptionError(
+      self, mock_time):
+    mock_time.return_value = 1401401409
+    mock_options = mock.Mock()
+    mock_options.length = 600
+    mock_options.ts_offset = 100
+    mock_options.step = 10
+    mock_options.update = True
+
+    self.assertRaises(
+        mlab_export.TimeOptionError, mlab_export.default_end_time, mock_options)
+
+  def testunit_default_start_time_WHEN_using_default_behavior(self):
+    # Default behavior is --update=True and a non-zero ts_previous value
+    # because the timestamp file aleady exists.
+    mock_options = mock.Mock()
+    mock_options.update = True
+    mock_options.step = 10
+    ts_previous = 1401401409
+    expected_start = 1401401400
+
+    returned_start = mlab_export.default_start_time(mock_options, ts_previous)
+
+    self.assertEqual(returned_start, expected_start)
+
+  def testunit_default_start_time_WHEN_first_run(self):
+    # On first run, ts_previous is zero b/c the timestamp file was just created.
+    mock_options = mock.Mock()
+    mock_options.update = True
+    mock_options.step = 10
+    mock_options.length = 3600
+    mock_options.ts_end = 1401401409 + mock_options.length
+    expected_start = 1401401400
+    ts_previous = 0
+
+    returned_start = mlab_export.default_start_time(mock_options, ts_previous)
+
+    self.assertEqual(returned_start, expected_start)
+
+  @mock.patch('mlab_export.logging.debug')
+  def testunit_assert_start_and_end_times(self, mock_logging):
+    mock_options = mock.Mock()
+    mock_options.length = 3600
+    mock_options.ts_end = 1401401409 + mock_options.length
```

Comment:
No problem. This is consistent with using the static timestamp throughout. fc7be16fe64f5b901e72fc503c5c902407a551f1